### ISSUE TITLE
Registration: User registration via link (ref code)

### ISSUE THREAD

**Issue by oyeaussie** [sp-framework/core#308]
Sometimes admins don't want to allow free registration on the site, instead they want to send out an invitation link. Option to configure that via add account. Admins can add a user and check which app allows user registration (with link) and send a link to the registered email address once account is added. 
Basically we are adding the account and forcing the user to change password via specific link.

link can be "register/q/link/iouqweroiuweopr"

where iouqweroiuweopr can be link base64 coded text and it can be stored in user table.

Apps need to have a configuration where they allow registration via link.

######## Issue imported from Gitea ########
# Details
Gitea Issue ID : 366
State : open
Created : 2023-04-16T11:21:45+10:00
# Issue Description
Sometimes admins don't want to allow free registration on the site, instead they want to send out an invitation link. Option to configure that via add account. Admins can add a user and check which app allows user registration (with link) and send a link to the registered email address once account is added. 
Basically we are adding the account and forcing the user to change password via specific link.

link can be "register/q/link/iouqweroiuweopr"

where iouqweroiuweopr can be link base64 coded text and it can be stored in user table.

Apps need to have a configuration where they allow registration via link.

# Timeline
Label : Added enhancement on 2023-04-16T11:21:45+10:00.
Label : Added P4 on 2023-04-16T11:21:45+10:00.
